### ISSUE TITLE
When orientation locked to landscape gridview renders firstly in portrait mode

### ISSUE THREAD

**Issue by kostyaBoss** [saleel/react-native-super-grid#23]
When orientation locked to landscape gridview renders firstly to portrait, then it  turns, but dimens are still as if screen is in portrait mode, then it extends to whole the screen and only then i can scroll them

**Comment by saleel** [comment]
I was not able to reproduce this. Can you tell me which OS you had this issue and post a screenshot please.

**Comment by kostyaBoss** [comment]
so i have two grid views in my component, wrapped by anvigation drawer. First is horizontal, and the second is not. By that quitely has no matter, if there is one or two of them. Also my component is locked to portrait orientation. The issue is in rendering, it comes step by step for my screen. It looks like smtnh like that. As far as i removing gridviews, rendering becomes as it must be. My OS is android 6.0.1 Xiaomi Redmi 3s PRO
![f81d0731-4dc2-4ed3-954a-b16b478a95c5](https://user-images.githubusercontent.com/20925367/35928996-226cfeae-0c37-11e8-8825-4f54783d87f3.jpg)
![c455800a-9d0b-469d-9849-3454a004425e](https://user-images.githubusercontent.com/20925367/35929022-2f1d662a-0c37-11e8-8d16-3f2a8d4166a5.jpg)

**Comment by kostyaBoss** [comment]
So, here is my code. I have a MainScreen.js component, which is implementing UI i should show on the main screen. Here is a code of this screen : 

    import React, {Component} from 'react';
    import {
        View,
        Text,
        StyleSheet,
        BackHandler,
        Alert,
        Platform,
        ToastAndroid,
        AppState,
        ImageBackground,
        Dimensions,
        TouchableOpacity,
        StatusBar
    }  from 'react-native';
    import { NavigationActions } from 'react-navigation';
    import Drawer from 'react-native-drawer';
    import Orientation from 'react-native-orientation';
    import MainGrid from './MainGrid'

    export default class MainScreen extends Component {

        static navigationOptions = {
            title: 'Меню',
            header : null
        }

        constructor(props) {
            super(props);

            this.state = {
                doubleBackToExitPressedOnce: false,
                appState: AppState.currentState,
                menuOpen: false,
           }
        }

        componentDidMount() {
            Orientation.lockToLandscape();

            BackHandler.addEventListener('hardwareBackPress', this.handleBackButton);
            // theese lines should log out user later
            // this.props.navigation.dispatch(resetAction)
            // ---------- const field to set route -----------------------
            // const resetAction = NavigationActions.reset({
            // index: 0,
            // actions: [
            //   NavigationActions.navigate({ routeName: 'MainScreen'})
            // ]
            // });
            // lock to lockToPortrait orientation
            AppState.addEventListener('change', this.handleAppStateChange);
        }

        componentWillUnmount() {
            BackHandler.removeEventListener('hardwareBackPress', this.handleBackButton);
            AppState.removeEventListener('change', this.handleAppStateChange);
        }

        handleAppStateChange = (nextAppState) => {
          if (this.state.appState.match(/inactive|background/) && nextAppState === 'active') {
            this.setState({doubleBackToExitPressedOnce: false})
          }
        this.setState({appState: nextAppState});
        }

        onButtonPress = () => {
            BackHandler.removeEventListener('hardwareBackPress', this.handleBackButton);
            // then navigate
            // navigate('NewScreen');
        }

        handleBackButton = () => {
            if (this.state.doubleBackToExitPressedOnce) {
                BackHandler.exitApp();
            }
            //ToastAndroid.show('Press back again to exit', ToastAndroid.SHORT);
            if (Platform.OS === 'ios') {
                Alert.alert(
                    'Выйти из приложения',
                    'Вы действительно желаете покинуть приложение? Все несохраненные данные будут         удалены', [{
                        text: 'Выйти',
                        onPress: () => {
                            console.log('Выйти pressed')
                            this.setState({
                                doubleBackToExitPressedOnce: true
                            });
                            BackHandler.exitApp();
                            return true;
                        }
                    },
                    {
                        text: 'Остаться',
                        onPress: () => {
                            console.log('Остаться pressed');
                            this.setState({
                                doubleBackToExitPressedOnce: false
                            });
                        },
                    }
                ], {
                    cancelable: false
                }
            )
            return true;
        } else {
            if (this.state.doubleBackToExitPressedOnce) {
                this.state.doubleBackToExitPressedOnce.setState
                BackHandler.exitApp();
            } else {
                ToastAndroid.show('Нажмите еще раз для выхода', ToastAndroid.SHORT);
                this.setState({
                    doubleBackToExitPressedOnce: true
                });
                return true;
            }
        }
    }
    closeControlPanel = () => {
      this._drawer.close()
    };
    openControlPanel = () => {
      this._drawer.open()
    };
    toggleDrawer() {
      this.state.toggled ? this._drawer.close() : this._drawer.open();
    }

      render() {
        return (
          <Drawer ref={(ref)=> this._drawer = ref}
             content={
               <View style={{backgroundColor: "rgba(255, 255, 255, 0.7)", height: 1000}}>
                 <StatusBar hidden = {true}/>
                 <Text>Hello</Text>
               </View>
            }
            type="overlay"
            openDrawerOffset={0.2}
            panCloseMask={0.4}
            closedDrawerOffset={-3}
            styles={drawerStyles}
            panOpenMask={0.05}
            captureGestures={true}
            tapToClose={true}>
              <ImageBackground
                resizeMode={'cover'}
                style={{ width: '100%', height: '100%' }}
                source={require('../../../images/coffee.jpg')}>
                  <MainGrid/>
              </ImageBackground>
         </Drawer>
       );
     }
    }

    const drawerStyles = {
      drawer: { shadowColor: '#fff', shadowOpacity: 0.8, shadowRadius: 3},
      main: {paddingLeft: 3},
    }

    const styles = StyleSheet.create({
        container: {
            flex: 1,
        },
    });

So this component, as you see locks an orientation to landscape mode, renders Navigation drawer and also MainGrid component, where all the greeds are been placed. Here is the code of Grids (MainGrid.js file) (both of them are in one component, sorry for my component styling right in code:))


    import React, { Component } from 'react';
    import {
      Platform,
      StyleSheet,
      Text,
      View,
      Dimensions,
      ImageBackground,
      ListView,
      TouchableOpacity,
      ToastAndroid,
      Image
    } from 'react-native';
    import Orientation from 'react-native-orientation-locker';
    import NavigationGrid from './NavigationGrid'
    import GridView from 'react-native-super-grid';


    export default class MainGrid extends Component {

      constructor(props) {
          super(props);
          this.state = {
              items : [],
              navigationItems : []
          }
      }
      componentDidMount()
      {
        for (let x = 1; x <= 30; x++) {
          this.state.items.push({
            name: `Grid ${x}`,
            price : `${x}$`
          });
        }
        for (let x = 1; x <= 10; x++) {
          this.state.navigationItems.push({
            name: `NavGrid ${x}`,
            price : `${x}$`
          });
        }
      }
     render() {

        return (
          <View style={styles.container}>
          <GridView
            style = {{height : Dimensions.get('window').height/5 + 90}}
            horizontal = {true}
            itemDimension={100}
            items={this.state.navigationItems}
            renderItem={item => (
              <TouchableOpacity
                style={styles.item}
                onPress = {() =>
                  ToastAndroid.show(item.name, ToastAndroid.SHORT)
                  }>
              <Image style={{borderRadius : 40,
                  height  : 80,
                  width : 80,
                  top: 0,
                  marginTop : 0,
                  marginLeft : 10,
                  position : 'absolute'}}
                source={require('../../../images/coffee.jpg')}/>
              <Text style ={{
                fontSize: 18,
                textAlign: 'center',
                width : '100%',
                color: '#fff',
                bottom: 0,
                marginBottom : 0,
                opacity: 0.5,
                position : 'absolute',
                backgroundColor: 'transparent',
              }} >{item.name}</Text>
              </TouchableOpacity>
            )}/>
          <GridView
            style = {{height : Dimensions.get('window').height - Dimensions.get('window').height/5 - 40}}
            itemDimension={100}
            items={this.state.items}
            renderItem={this.renderGridItem}/>
          </View>
        );
      }

      renderGridItem = (item) => (
          <TouchableOpacity
            style={styles.item}
            onPress = {() =>
              ToastAndroid.show(item.name, ToastAndroid.SHORT)
              }>
            <ImageBackground style={{flex : 1}}
            source={require('../../../images/coffee.jpg')}/>
            <Text style={styles.name}>
              {item.name}
            </Text>
            <Text style={styles.price}>
              {item.price}
            </Text>
          </TouchableOpacity>
        );
    }



    const styles = StyleSheet.create({
      container: {
        flex: 4,
        display: 'flex',
        flexDirection : 'column',
        marginTop: 20,
        backgroundColor: 'transparent',
      },
      welcome: {
        fontSize: 20,
        textAlign: 'center',
        margin: 10,
      },
      flex: {
        flex: 1,
        flexDirection: 'column',
      },
      item: {
        height: 100,
        backgroundColor : 'rgba(200, 0, 0, 0.6)',
        width : 100,
        marginRight : 16,
        padding: 0,
      },
      name: {
        fontSize: 14,
        textAlign: 'left',
        position : 'absolute',
        width : '100%',
        color: '#000000',
        bottom: 0,
        paddingLeft : 10,
        marginBottom : 0,
        opacity: 0.5,
        backgroundColor: 'rgba(255, 255, 255, 0.7)',
      },
      price : {
        fontSize: 14,
        width : '100%',
        position : 'absolute',
        textAlign: 'right',
        marginBottom : 0,
        bottom: 0,
        paddingRight : 5,
        color: 'red',
      }
    })

I've also tried to remove grids, then it renders quite fast:)
Tried not to lock orientation, but grids are still rendering as i described on picture below, but just in portrait orientation

Any ideas?

**Comment by saleel** [comment]
I tried this code in Redmi Note 4. Everything working fine when the orientation lock is commented.

When its uncommented, I am getting error.

I think this is out of the scope of this component. It could be an issue with the orientation component you are using.

I tried the OS orientation lock and everything is normal as well.

![screenshot_2018-02-09-23-37-36-619_com griddemo](https://user-images.githubusercontent.com/13466197/36046549-bb215afc-0df2-11e8-8a9d-b489f382d4ec.png)
